manager/manager.py

```diff
@@ -0,0 +1,1105 @@
+from functools import wraps
+from pathlib import Path
+from types import SimpleNamespace
+from typing import Type, Union
+
+import appdirs
+import paramiko
+import yaml
+from ftpsync.ftp_target import FTPTarget
+from ftpsync.sftp_target import SFTPTarget
+from ftpsync.synchronizers import DownloadSynchronizer, UploadSynchronizer
+from ftpsync.targets import FsTarget
+
+# --------------------------------------------------------------------------------------------------------------------
+# Folder Class
+# --------------------------------------------------------------------------------------------------------------------
+
+
+class Folder:
+    """
+    Folder class used to contain details of canonical directories in the project directory tree.
+    """
+
+    def __init__(self, name, used, subfolders=None):
+        self.name = name
+        self.used = used
+        self.subfolders = subfolders
+
+
+# --------------------------------------------------------------------------------------------------------------------
+# Class Decorators
+# --------------------------------------------------------------------------------------------------------------------
+
+
+def requires_ssh_configs(func):
+    """
+    Decorator to check file is loaded. Used on Mainwindow class methods only as first arg is assumed to be self (containing cfgs)
+    """
+
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        if not args[0].cfg["remote_host_id"] or not args[0].cfg["remote_host_username"]:
+            args[0]._raise_error(
+                "Cannot setup SSH connection, configuration file remote_host_id or remote_host_username is not set."
+            )
+        else:
+            return func(*args, **kwargs)
+
+    return wrapper
+
+
+# --------------------------------------------------------------------------------------------------------------------
+# Project Manager Class
+# --------------------------------------------------------------------------------------------------------------------
+
+
+class ProjectManager:
+    """
+    Main project manager class for data organisation and transfer in BIDS-style project directory.
+    The expected organisation is a central repository on a linux remote cluster ('remote') that
+    contains all project data. This is connected to multiple local machines ('local') which
+    each contain a subset of the full project (e.g. machine for electrophysiology collection,
+    machine for behavioural connection, machine for analysis for specific data etc.).
+
+    On first use on a new profile, the user will be prompted to set configurations with the function
+    make_config_file().
+
+    For transferring data between a remote data storage with SSH, use setup setup_ssh_connection_to_remote_server().
+    This will allow you to check the server Key, add host key to profile if accepted, and setup ssh key pair.
+
+    INPUTS: username - the username for the profile to use. Profile files are stored in your Appdir folder
```

Comment:
It may be worth specifying for the user what username is this. It's not immediately clear whether this is for the local machine, the remote server, or maybe even for this software itself.

Reply:
edited username was not a great description, project_name more appropriate as for each project a new config must be setup (including root dir tree etc).